### PR TITLE
Use hex instead of hash interface in partMetadata

### PR DESCRIPTION
**Pull request by vadmeste**
It would be little easier to use hex directly to express
a part's md5 or sha256 sum instead of hash interface. Convert
hex to base64 whenever we need that.

### CODE REVIEW

File: minio/api.py [hunk 1]
```diff
@@ -1558,7 +1559,7 @@ def _do_put_multipart_object(self, bucket_name, object_name, part_metadata,
 
         headers = {
             'Content-Length': part_metadata.size,
-            'Content-Md5': part_metadata.md5hasher.base64digest()
+            'Content-Md5': base64.b64encode(part_metadata.md5_hex.decode("hex"))
```
Comment (poornas) [inline minio/api.py]: For python3 , this does not work - I get an AttributeError: 'str' object has no attribute 'decode' part_metadata.md5_hex  is of type string.

Comment (vadmeste) [inline minio/api.py]: > For python3 , this does not work - I get an AttributeError: 'str' object has no attribute 'decode' part_metadata.md5_hex is of type string.

Weird, I didn't get any error when I tested `python3 examples/fput_object.py`, can you tell me how did you test exactly ?

Comment (poornas) [inline minio/api.py]: The error happens if you run python3 fput_object.py, i.e. by moving the fput_object.py from examples/ folder to the main repo directory.

Comment (harshavardhana) [inline minio/api.py]: Basically this would import your current code @vadmeste

Comment (vadmeste) [inline minio/api.py]: @poornas, the issue is fixed, can you check/review again ?

**Comment by vadmeste** [comment]
@harshavardhana can you review this again and merge?

**Pull request merged**